repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/DataProductPreProcessor.java

```diff
@@ -443,6 +442,9 @@ public void processDelete(AtlasVertex vertex) throws AtlasBaseException {
                     }
                 }
             }
+            if(RequestContext.get().getDeleteType() == DeleteType.SOFT){
```

Comment:
simply adding else here will have the same flow IMO

Reply:
In both cases i.e SOFT and PURGE the else will get called hence added a if condition